bftengine/src/preprocessor/RequestProcessingInfo.cpp

```diff
@@ -69,20 +69,35 @@ auto RequestProcessingInfo::calculateMaxNbrOfEqualHashes(uint16_t &maxNumOfEqual
   return itOfChosenHash;
 }
 
+// Primary replica logic
 bool RequestProcessingInfo::isReqTimedOut() const {
   // Check request timeout once asynchronous primary pre-execution completed (to not abort the execution thread)
   if (primaryPreProcessResultLen_ != 0) {
     auto reqProcessingTime = getMonotonicTimeMilli() - entryTime_;
     if (reqProcessingTime > clientPreProcessReqMsg_->requestTimeoutMilli()) {
       LOG_WARN(GL,
                "Request timeout of " << clientPreProcessReqMsg_->requestTimeoutMilli() << " ms expired for reqSeqNum="
-                                     << reqSeqNum_ << "; reqProcessingTime=" << reqProcessingTime << "; abort request");
+                                     << reqSeqNum_ << "; reqProcessingTime=" << reqProcessingTime);
       return true;
     }
   }
   return false;
 }
 
+// Non-primary replica logic
+bool RequestProcessingInfo::isPreProcessReqMsgReceivedInTime(const uint16_t preProcessReqWaitTimeMilli) const {
```

Comment:
Hi Yulia - I couldn't find where you call `isPreProcessReqMsgReceivedInTime` ?
My guess is - the PR is still WIP, so everything is not fully wired yet.

Reply:
Yes, exactly :-)